front/app/modules/commercial/project_folders/admin/containers/permissions/index.tsx

```diff
@@ -58,12 +56,11 @@ const UserSelectButton = styled(Button)`
   margin-left: 12px;
 `;
 
-const FolderPermissions = ({ intl: { formatMessage } }: InjectedIntlProps) => {
-  const { projectFolderId } = useParams();
+const FolderPermissions = ({
```

Comment:
Instead of reverting updates that we do want to see in the codebase (moving away from `withRouter`) we should come up with a good pattern for these cases. What if we gave the param a default value (`""`) for example? Then there is no need to call the rest of the hooks conditionally.

Reply:
Fully agree that we should move away from withRouter but I think if we need to come up with a completely new pattern it should happen outside the scope of this PR. I don't think `const projectFolderId = useParams().projectFolderId || ""` is necessarily more clear or easier to reason about, even if it does fix the linter issue.

In this case I think the injected `withRouter` props are actually more accurate than `useParams` as far as types go. To get to this component, the route is `path: 'projects/folders/:projectFolderId/permissions'`, so as long as the route is matched there should always be a string returned for `params.projectFolderId`. The `useParams` hook is typed to return either a `string` or `undefined` so in order to pass the ID in to our `useProjectFolderModerators(projectFolderId: string)` hook, which only accepts a string, you have to either do a null check (which doesn't play well with hook order) or assign an empty string if the param is undefined, which I don't think should be possible based on how the routing is set up. 

From checking the codebase I count ~50 instances of getting params from the injected `withRouter` props and then passing them directly into a hook without any sort of null check, so if we want to replace this pattern I think we should consider our options a bit more, and in the interests of keeping the changes in this already massive PR as minimal as possible, do that as a follow-up rather than adding more scope here. If that makes sense I'm happy to create a follow-up ticket for this.

Reply:
Good arguments! Agree it's probably out of scope here. But let's add a ticket and have a discussion about what's the preferred way to access params moving forward :)

Reply:
Solid :) https://citizenlab.atlassian.net/browse/TEC-187 is the ticket, I imagine we'll need to discuss this relatively soon after the react-router changes are merged since we'll start writing components in this way. In the meantime we have a bit of time to consider the various options